src/GitTfs.VsCommon/TfsHelper.Common.cs

```diff
@@ -649,13 +649,20 @@ protected ITfsChangeset BuildTfsChangeset(Changeset changeset, IGitTfsRemote rem
 
             if (HasWorkItems(changeset))
             {
-                tfsChangeset.Summary.Workitems = changeset.WorkItems.Select(wi => new TfsWorkitem
+                try
+                {
+                    tfsChangeset.Summary.Workitems = changeset.WorkItems.Select(wi => new TfsWorkitem
+                    {
+                        Id = wi.Id,
+                        Title = wi.Title,
+                        Description = wi.Description,
+                        Url = Linking.GetArtifactUrl(wi.Uri.AbsoluteUri)
+                    });
+                }
+                catch (TeamFoundationServerException e)
```

Comment:
I am currious, do you remember which property caused the exception? At least the code in HasWorkItems method suggests that the WorkItems array can be fetched.

Please check for the concrete error TF201077, to ensure we are not accidentially ignoring other errors.
See e.g.  the code in `HasWorkItems` a few lines below, how to check the exception message method, inlined here
```csharp
        protected virtual bool HasWorkItems(Changeset changeset)
        {
            // This method wraps changeset.WorkItems, because
            // changeset.WorkItems might result to ConnectionException: TF26175: Team Foundation Core Services attribute 'AttachmentServerUrl' not found.
            // in this case assume that it is initialized to null
            // NB: in VS2011 a new property appeared (AssociatedWorkItems), which works correctly
            WorkItem[] result = null;
            try
            {
                result = Retry.Do(() => changeset.WorkItems);
            }
            catch (ConnectionException exception)
            {
                if (!exception.Message.StartsWith("TF26175:"))
                    throw;
            }
            return result != null && result.Length > 0;
        }
```

Reply:
So I've refactored the code a bit using both AssociatedWorkItems (which works even for malformed work items) and WorkItems to update the TfsWorkItem instances (which can cause an exception).